src/R/Components/Impl/InteractiveWorkflow/Implementation/RSessionCallback.cs

```diff
@@ -60,37 +62,30 @@ internal sealed class RSessionCallback : IRSessionCallback {
         /// Displays R plot in the host app-provided window
         /// </summary>
         public async Task Plot(PlotMessage plot, CancellationToken ct) {
-            var provider = _coreShell.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
-            var workflow = provider.GetOrCreate();
+            var workflow = _workflowProvider.GetOrCreate();
 
             await _coreShell.SwitchToMainThreadAsync();
             await workflow.Plots.LoadPlotAsync(plot);
         }
 
         public async Task<LocatorResult> Locator(Guid deviceId, CancellationToken ct) {
-            var provider = _coreShell.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
-            var workflow = provider.GetOrCreate();
+            var workflow = _workflowProvider.GetOrCreate();
 
             await _coreShell.SwitchToMainThreadAsync();
-
             return await workflow.Plots.StartLocatorModeAsync(deviceId, ct);
         }
 
         public async Task<PlotDeviceProperties> PlotDeviceCreate(Guid deviceId, CancellationToken ct) {
-            var provider = _coreShell.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
-            var workflow = provider.GetOrCreate();
+            var workflow = _workflowProvider.GetOrCreate();
 
             await _coreShell.SwitchToMainThreadAsync();
-
             return await workflow.Plots.DeviceCreatedAsync(deviceId);
         }
 
         public async Task PlotDeviceDestroy(Guid deviceId, CancellationToken ct) {
-            var provider = _coreShell.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
-            var workflow = provider.GetOrCreate();
+            var workflow = _workflowProvider.GetOrCreate();
```

Comment:
If `_workflowProvider` is disposed already, it will throw `ObjectDisposedException`. Do we handle it anywhere?